lib/fluent/plugin/filter_filter_list.rb

```diff
@@ -1,21 +1,24 @@
 require 'fluent/plugin/out_filter_list/version'
-require 'aho_corasick'
+require 'matcher'
 require 'fluent/plugin/filter'
+require 'ip'
 
 module Fluent
   module Plugin
     class FilterListFilter < Filter
       include Matchers
+      include IP
 
       Plugin.register_filter('filter_list', self)
 
+      config_param :filter, :string, default: 'AC'
       config_param :key_to_filter, :string, default: nil
       config_param :patterns_file_path, :string, default: ''
 
       def configure(conf)
         super
         patterns = @patterns_file_path.empty? ? [] : File.readlines(@patterns_file_path).map(&:chomp).reject(&:empty?)
-        @matcher = ACMatcher.new(patterns)
+        @matcher = (@filter == 'AC') ? ACMatcher.new(patterns) : IPMatcher.new(patterns)
```

Comment:
I'd like the default matcher to be the _AC_. So when `@filter` is unknown, please make `@matcher` an instance of `ACMatcher`.

Reply:
I think `@filter`'s default value must be `AC` in line 14.